### PR TITLE
Removing req_amount from crafting recipes for items and structures.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
All the important parts of the original PR got split off into their own PRs so I'm going to use it for this small one. Removes preset amounts from item/structure recipes so they will autogenerate from matter.

### CODE REVIEW

File: code/game/machinery/doors/simple.dm [hunk 1]
```diff
@@ -73,13 +70,15 @@
 /obj/machinery/door/unpowered/simple/close(var/forced = 0)
 	if(!can_close(forced))
 		return
-	playsound(src.loc, material.dooropen_noise, 100, 1)
+	if(material)
```
Comment (comma) [inline code/game/machinery/doors/simple.dm]: not sure if there's any point checking for material considering it frekaing qdels self if it hasn't got one

File: code/game/machinery/kitchen/smartfridge.dm [hunk 2]
```diff
@@ -241,10 +241,11 @@
 	overlays += I
 
 /obj/machinery/smartfridge/dismantle()
-	for(var/datum/stored_items/I in item_records)
-		while(I.amount > 0)
-			I.get_product(get_turf(src)) // They'd get dumped anyway, but this makes things GC properly.
-	..()
+	if(!dismantled)
```
Comment (comma) [inline code/game/machinery/kitchen/smartfridge.dm]: isn't this var set in the base call, why check it first?

File: code/game/objects/obj_materials.dm [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		desc = base_desc
+
+/obj/proc/update_material_colour(var/override_colour)
+	var/base_colour = override_colour || initial(color)
+	if(istype(material))
+		color = material.icon_colour
+	else
+		color = base_colour
+
+/obj/on_update_icon()
```
Comment (comma) [inline code/game/objects/obj_materials.dm]: I think a /lot/ of on_update_icon() calls don't call parent, I'm like 40% that not worrying about what parent might have was the point of making it a separate proc. 
Maybe it's something that should go into obj update_icon() call so it's reliably called?

Comment (MistakeNot4892) [reply code/game/objects/obj_materials.dm]: Well it'd give update_icon() an actual purpose at least I guess.

**Comment by comma** [comment]
gib generation results

**Comment by MistakeNot4892** [comment]
[message (29).txt](https://github.com/NebulaSS13/Nebula/files/4519676/message.29.txt)

**Review by afterthought2 — DISMISSED** [review]

**Review by afterthought2 — APPROVED** [review]

**Pull request merged**